exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -23,7 +22,7 @@ import kotlin.reflect.full.isSubclassOf
 /**
  * Interface common to all column types.
  */
-interface IColumnType {
+interface IColumnType<T> {
```

Comment:
This is the starting point. The goal is to have type safety in the overridden functions, which will reduce the number of checks in those functions.